python/ray/_private/resource_isolation_config.py

```diff
@@ -0,0 +1,191 @@
+import logging
+from typing import Optional
+
+import ray._private.ray_constants as ray_constants
+
+import ray._private.utils as utils
+
+logger = logging.getLogger(__name__)
+
+# See https://docs.kernel.org/admin-guide/cgroup-v2.html#weights
+# for information about cpu weights
+_CGROUP_CPU_MAX_WEIGHT: int = 10000
+
+
+class ResourceIsolationConfig:
+    """Configuration for enabling resource isolation by reserving memory
+    and cpu for ray system processes through cgroupv2.
+    This class validates configuration for resource isolation by
+    enforcing types, correct combinations of values, applying default values,
+    and sanity checking cpu and memory reservations.
+    Also, converts system_reserved_cpu into cpu.weights for cgroupv2.
+
+    Attributes:
+        enable_resource_isolation: True if cgroupv2 based isolation of ray
+            system processes is enabled.
+        cgroup_path: The path for the cgroup the raylet should use to enforce
+            resource isolation.
+        system_reserved_cpu: The amount of cores reserved for ray system
+            processes.
+        system_reserved_memory: The amount of memory in bytes reserved
+            for ray system processes.
+    """
+
+    def __init__(
+        self,
+        enable_resource_isolation: bool = False,
+        cgroup_path: Optional[str] = None,
+        system_reserved_cpu: Optional[float] = None,
+        system_reserved_memory: Optional[int] = None,
+    ):
+
+        self.enable_resource_isolation = enable_resource_isolation
+        self.cgroup_path = cgroup_path
+        self.system_reserved_cpu = system_reserved_cpu
+        self.system_reserved_memory = system_reserved_memory
+        self.system_reserved_cpu_weight: float = None
+        self.constructed = False
+
+        if not enable_resource_isolation:
```

Comment:
for readability purposes you can consider splitting some of the validation logic into private methods